managedplugin/plugin.go

```diff
@@ -686,6 +687,51 @@ func (c *Client) Versions(ctx context.Context) ([]int, error) {
 	return res, nil
 }
 
+func (c *Client) FindLatestPluginVersion(ctx context.Context, typ PluginType) (string, error) {
+	if c.config.Registry != RegistryCloudQuery {
+		return "", fmt.Errorf("plugin registry is not cloudquery; cannot find latest plugin version")
+	}
+
+	pathSplit := strings.Split(c.config.Path, "/")
+	if len(pathSplit) != 2 {
+		return "", fmt.Errorf("invalid cloudquery plugin path: %s. format should be team/name", c.config.Path)
+	}
+	org, name := pathSplit[0], pathSplit[1]
+
+	ops := HubDownloadOptions{
+		AuthToken:     c.authToken,
+		TeamName:      c.teamName,
+		LocalPath:     c.LocalPath,
+		PluginTeam:    org,
+		PluginKind:    typ.String(),
+		PluginName:    name,
+		PluginVersion: c.config.Version,
+	}
+	hubClient, err := getHubClient(c.logger, ops)
+	if err != nil {
+		return "", fmt.Errorf("failed to get hub client: %w", err)
+	}
+
+	if ops.TeamName == "" {
```

Comment:
Maybe check `c.TeamName` (as early as possible, around where `c.config.Registry` is checked) directly so it's easier to read.

Reply:
I checked it at the top now.